src/CacheItem.php

```diff
@@ -44,20 +49,20 @@ class CacheItem implements HasExpirationDateInterface, CacheItemInterface, Tagga
     private $hasValue = false;
 
     /**
-     * @param string                  $key
-     * @param bool                    $hasValue
-     * @param mixed                   $value
-     * @param \DateTimeInterface|null $expirationDate
+     * @param string        $key
+     * @param \Closure|bool $callable or boolean hasValue
      */
-    public function __construct($key, $hasValue = false, $value = null, \DateTimeInterface $expirationDate = null)
+    public function __construct($key, $callable = null, $value = null)
     {
-        $this->taggedKey      = $key;
-        $this->key            = $this->getKeyFromTaggedKey($key);
-        $this->hasValue       = $hasValue;
-        $this->expirationDate = $expirationDate;
-
-        if ($hasValue) {
-            $this->value = $value;
+        $this->taggedKey = $key;
+        $this->key       = $this->getKeyFromTaggedKey($key);
+
+        if ($this->callable === true) {
+            $this->hasValue = true;
+            $this->value    = $value;
+        } elseif ($this->callable !== false) {
+            // This must be a callable or null
+            $this->callable = $callable;
         }
```

Comment:
I want to allow `new CacheItem($key, true, $value)`. This would be better if you run getItems() on some storages.